### ISSUE TITLE
PostgreSQL: L029 vs. L059

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3052]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Contradicting errors occur for aliases that are keywords.

### Expected Behaviour

After receiving the following error:
```
L:   1 | P:  18 | L029 | Keywords should not be used as identifiers.
```

Quoting that keyword alias solves the issue.

### Observed Behaviour

Got a new error:
```
L:   2 | P:  18 | L059 | Unnecessary quoted identifier "name".
```

### How to reproduce

```sql
SELECT 'name' AS name;
SELECT 'name' AS "name";
```

### Dialect

postgres

### Version

sqlfluff, version 0.12.0

### Configuration

default (option `--dialect postgres` given to CLI invocations)

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
These are somewhat contradictory I agree, but they are checking two different things (even if it seems they are related!).

L029 says you shouldn't used ANY keywords - including unreserved keywords that CAN be used, but the rule advises should not be used. You can quote them to try to get around this rule (as by default `quoted_identifiers_policy = none` for L029 but you can change that if you want), but quotes are not needed for unreserved keywords, so that's not necessary, so not the right answer, and leads to an L059 flagging instead (which I'll cover below).

Your options are:
1. Don't use unreserved keywords, even though you can. That's what this rule is highlighting, not the fact you've used them in the incorrect way.
2. Turn off L029 if you don't find it that useful (I don't - reserved keywords are not allowed in unquoted format anyway and restricting unreserved keywords isn't necessary).
3. Leave rule L029 on if you agree with it in general, even for unreserved words, but add the ones you are happy to use to the `ignore_words` config for this rule. This works well if you want to use a few unreserved keywords but restrict the rest.
4. Leave L029 in place in full, but add `noqa: L029` in SQL if you are OK with that specific instance.

L059 says only use quotes when you need to. In this case you don't need to as `name` is an unreserved keyword. Again you have similar options to above to avoid an error if you do want to quote this (perhaps to get around L029).

So while I agree it's somewhat confusing, it is working as intended. You need to decide what policy you want to follow on these and configure L029 and L059 as appropriate, or just not use unreserved keywords.

**Comment by enote-kane** [comment]
@tunetheweb Thank you for taking the time to disassemble and clarify that for me. Highly appreciated!
